packages/openapi-generator/src/parser/schema.ts

```diff
@@ -59,13 +51,28 @@ export function parseSchema(
   }
 
   if (schema.allOf?.length) {
+    if (schema.properties) {
```

Comment:
IDEA: I am not sure what else could lie next to `allOf`. Wouldn't it be nice to "normalize" the schema through something along the lines of this:
```ts
const { allOf, ...objectSchema } = schema;
return parseXOfSchema({allOf: [...allOf, objectSchema] }, ref, 'allOf', options);
```

(I think you mentioned `required` could also be next to allOf, but then just add that as well.)

Reply:
@marika In case where `allOf` and `properties` are at the same level (and have required), this actually works quite well.
For the other case (which is the case in the spec):

```
allOf:
  - ref
  - type: object //there can be multiple of such objects i imagine
     properties:
       - ...
required:
 -
```

Here I would need to add `required` to(probably) every object item. I had handled this in `parseXOfSchema` before as:

```
return {
    [xOf]: (schema[xOf] || []).map(entry =>
      parseSchema({ ...entry, required }, refs, options)  };
```

Wouldn't this mean the same thing?

Reply:
Updated it now (incl. merging the required properties from different levels). Please have a look